src/state_machine/signer/mod.rs

```diff
@@ -284,17 +334,44 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
                     signer_id: self.signer_id,
                     status: DkgStatus::Success,
                 },
-                Err(dkg_error_map) => DkgEnd {
-                    dkg_id: self.dkg_id,
-                    signer_id: self.signer_id,
-                    status: DkgStatus::Failure(format!("{:?}", dkg_error_map)),
-                },
+                Err(dkg_error_map) => {
+                    // we've handled everything except BadShares and Point both of which should map to DkgFailure::BadPrivateShares
+                    let mut bad_private_shares = HashMap::new();
+                    for (_my_party_id, dkg_error) in dkg_error_map {
+                        match dkg_error {
```

Comment:
Similar to the other nit, I would do an if else rather than a match since its a single match arm (unless the todo expands the match arm)

Reply:
Yeah I'm surprised clippy let so many of these through.  If nothing else, getting rid of the match reduces the indentation by one level, which is a win by itself.

Reply:
Done.